src/item.cpp

```diff
@@ -2371,6 +2376,35 @@ bool item::has_flag( const std::string &f ) const
     return ret;
 }
 
+bool item::has_property (const std::string& prop) const {
+   return type->properties.find(prop) != type->properties.end();
+}
+
+std::string item::get_property (const std::string &prop, const std::string& def) const
```

Comment:
It makes perfect sense for the method to infer that it should return a type
matching a provided default value, but it makes no sense to require the
caller to provide an extraneous default value to tell the method what type
to return.